apps/lockfile-explorer-web/src/parsing/lfxGraphLoader.ts

```diff
@@ -0,0 +1,363 @@
+
+      const underscoreIndex = versionPart.indexOf('_');
+      if (underscoreIndex >= 0) {
+        const version = versionPart.substring(0, underscoreIndex);
+        const suffix = versionPart.substring(underscoreIndex + 1);
+
+        result.entryPackageVersion = version;
+        result.entrySuffix = suffix;
+
+        //       /@rushstack/eslint-config/3.0.1_eslint@8.21.0+typescript@4.7.4
+        // -->   @rushstack/eslint-config 3.0.1 (eslint@8.21.0+typescript@4.7.4)
+        result.displayText = packageName + ' ' + version + ' (' + suffix + ')';
+      } else {
+        result.entryPackageVersion = versionPart;
+
+        //       /@rushstack/eslint-config/3.0.1
+        // -->   @rushstack/eslint-config 3.0.1
+        result.displayText = packageName + ' ' + versionPart;
+      }
+    }
+
+    // Example:
+    //   common/temp/default/node_modules/.pnpm
+    //     /@babel+register@7.17.7_@babel+core@7.17.12
+    //     /node_modules/@babel/register
+    result.packageJsonFolderPath =
+      `common/temp/${subspaceName}/node_modules/.pnpm/` +
+      result.entryPackageName.replace('/', '+') +
+      '@' +
+      result.entryPackageVersion +
+      (result.entrySuffix ? `_${result.entrySuffix}` : '') +
+      '/node_modules/' +
+      result.entryPackageName;
+  }
+
+  parseDependencies(result.dependencies, result, rawYamlData);
+
+  return result;
+}
+
+/**
+ * Transform any newer lockfile formats to the following format:
+ * [packageName]:
+ *     specifier: ...
+ *     version: ...
+ */
+function getImporterValue(
+  importerValue: ILockfileImporterV5 | ILockfileImporterV6,
+  pnpmLockfileVersion: PnpmLockfileVersion
+): ILockfileImporterV5 {
+  if (pnpmLockfileVersion === PnpmLockfileVersion.V6) {
+    const v6ImporterValue = importerValue as ILockfileImporterV6;
+    const v5ImporterValue: ILockfileImporterV5 = {
+      specifiers: {},
+      dependencies: {},
+      devDependencies: {}
+    };
+    for (const [depName, depDetails] of Object.entries(v6ImporterValue.dependencies || {})) {
+      v5ImporterValue.specifiers![depName] = depDetails.specifier;
+      v5ImporterValue.dependencies![depName] = depDetails.version;
+    }
+    for (const [depName, depDetails] of Object.entries(v6ImporterValue.devDependencies || {})) {
+      v5ImporterValue.specifiers![depName] = depDetails.specifier;
+      v5ImporterValue.devDependencies![depName] = depDetails.version;
+    }
+    return v5ImporterValue;
+  } else {
+    return importerValue as ILockfileImporterV5;
+  }
+}
+
+/**
+ * Parse through the lockfile and create all the corresponding LockfileEntries and LockfileDependencies
+ * to construct the lockfile graph.
+ *
+ * @returns A list of all the LockfileEntries in the lockfile.
+ */
+export function generateLockfileGraph(lockfile: ILockfilePackageType, subspaceName?: string): LfxGraph {
+  let pnpmLockfileVersion: PnpmLockfileVersion = PnpmLockfileVersion.V5;
+  if (`${lockfile.lockfileVersion}`.startsWith('6')) {
```

Comment:
```suggestion
  if (parseInt(lockfile.lockfileVersion) === 6) {
```

Reply:
He's trying to also check for `6.1` (but apparently not concerned about `60.0`)


We'll clean this up in a later PR.

Reply:
Ohh of course, it's JavaScript.  😆 I thought `parseInt()` would reject strings that aren't integers haha

Reply:
Fixed